### ISSUE TITLE
IsExternalInit compiler error on STJ.7.0.3 in SBRP

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#4159]
When regenerating the SBRP source for System.Text.Json.7.0.3, it changes some of the properties to use `init` instead of `set` (see https://github.com/dotnet/source-build-reference-packages/pull/896/commits/6f570e9dccbaeb8e24fe4f1d1552eb92d5fdbbc7#diff-169c8172108e072ea318044927600380ad8d3750793a9af77997f74a0b307f6c). This change causes a compiler error:

```
error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
```

**Comment by MichaelSimons** [comment]
genAPI issue?  

cc @ViktorHofer

**Comment by ViktorHofer** [comment]
cc @ericstj

**Comment by mthalman** [comment]
Yeah, I wasn't sure if it was GenAPI or if things aren't configured correctly in the SBRP build infrastructure.

**Comment by ericstj** [comment]
GenAPI - please manually include the definition of IsExternalInit in the sources for frameworks which don't expose that type (netstandard2.0 here).  It's defined internally in the source assembly.  This is the same issue as https://github.com/dotnet/sdk/issues/33693.
